presto-main/src/main/java/com/facebook/presto/operator/MoreByteArrays.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.api.Experimental;
+
+import static com.facebook.presto.spi.JvmUtils.unsafe;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static java.util.Objects.requireNonNull;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+
+@Experimental
+public class MoreByteArrays
+{
+    public static byte getByte(byte[] bytes, int index)
+    {
+        checkPositionIndex(index, SIZE_OF_BYTE, bytes.length);
+        return unsafe.getByte(bytes, (long) index + ARRAY_BYTE_BASE_OFFSET);
+    }
+
+    public static int fill(byte[] bytes, int index, int length, byte value)
+    {
+        requireNonNull(bytes, "bytes is null");
+        checkPositionIndex(index, length, bytes.length);
+
+        unsafe.setMemory(bytes, index + ARRAY_BYTE_BASE_OFFSET, length, value);
+        return index + length;
+    }
+
+    public static int setBytes(byte[] bytes, int index, byte[] values, int offset, int length)
```

Comment:
nit: What about name parameters as
```
byte[] destination, int destOffset, byte[] source, int sourceOffest, int length
```

The name `source` and `destination` is inspired by `arraycopy` in Java and `memcpy` in C.

Reply:
Hi @wenleix the naming here is to mimic the ones in in ByteArrays in Airlift. For example

```
    public static void setShort(byte[] bytes, int index, short value)
    {
        checkIndexLength(bytes.length, index, SIZE_OF_SHORT);
        unsafe.putShort(bytes, ((long) ARRAY_BYTE_BASE_OFFSET) + index, value);
    }
```
Since we named the whole class as MoreByteArrays should we keep the names aligned with ByteArrays?